test/test_proxy.py

```diff
@@ -0,0 +1,298 @@
+
+    @staticmethod
+    def get_tls_connection_options_for_test(test_type, host_name):
+        if test_type == ProxyTestType.FORWARDING or test_type == ProxyTestType.LEGACY_HTTP or test_type == ProxyTestType.TUNNELING_HTTP:
+            return None
+        else:
+            tls_ctx_opt = TlsContextOptions()
+            tls_ctx = ClientTlsContext(tls_ctx_opt)
+            tls_conn_opt = tls_ctx.new_connection_options()
+            tls_conn_opt.set_server_name(host_name)
+            return tls_conn_opt
+
+    @staticmethod
+    def get_uri_from_test_type(test_type):
+        if test_type == ProxyTestType.FORWARDING or test_type == ProxyTestType.LEGACY_HTTP or test_type == ProxyTestType.TUNNELING_HTTP:
+            return "www.example.com"
+        else:
+            return "www.amazon.com"
+
+    @staticmethod
+    def get_port_from_test_type(test_type):
+        if test_type == ProxyTestType.FORWARDING or test_type == ProxyTestType.LEGACY_HTTP or test_type == ProxyTestType.TUNNELING_HTTP:
+            return 80
+        else:
+            return 443
+
+
+class ProxyHttpTest(NativeResourceTest):
+
+    timeout = 10  # seconds
+
+    def _establish_http_connection(self, test_type, uri, proxy_options):
+        event_loop_group = EventLoopGroup()
+        host_resolver = DefaultHostResolver(event_loop_group)
+        bootstrap = ClientBootstrap(event_loop_group, host_resolver)
+
+        connection_future = HttpClientConnection.new(
+            host_name=uri,
+            port=ProxyTestConfiguration.get_port_from_test_type(test_type),
+            bootstrap=bootstrap,
+            tls_connection_options=ProxyTestConfiguration.get_tls_connection_options_for_test(
+                test_type,
+                uri),
+            proxy_options=proxy_options)
+        return connection_future.result(self.timeout)
+
+    def _do_proxy_http_test(self, test_type, auth_type):
+        uri = ProxyTestConfiguration.get_uri_from_test_type(test_type)
+        proxy_options = ProxyTestConfiguration.create_http_proxy_options_from_environment(test_type, auth_type)
+        connection = self._establish_http_connection(test_type, uri, proxy_options)
+
+        request = HttpRequest('GET', '/')
+        request.headers.add('host', uri)
+        response = Response()
+        stream = connection.request(request, response.on_response, response.on_body)
+        stream.activate()
+
+        # wait for stream to complete
+        stream_completion_result = stream.completion_future.result(self.timeout)
+
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(200, stream_completion_result)
+
+        return
+
+    @unittest.skipIf(not ProxyTestConfiguration.is_proxy_environment_initialized(), 'requires proxy test env vars')
+    def test_forwarding_proxy_no_auth(self):
+        self._do_proxy_http_test(ProxyTestType.FORWARDING, HttpProxyAuthenticationType.Nothing)
+        return
```

Comment:
nitpick: delete all these blank `return` statements. they're not needed, and they look weird